src/telliot_feeds/reporters/tips/listener/funded_feeds_filter.py

```diff
@@ -0,0 +1,194 @@
+        return feeds
+
+    def timestamps_filter(self, feeds: List[QueryIdandFeedDetails]) -> List[QueryIdandFeedDetails]:
+        for feed in feeds:
+            if len(feed.queryid_timestamps_list) < 2:
+                continue
+            for i, _ in enumerate(list(feed.queryid_timestamps_list)):
+                in_eligibile_window = self.is_timestamp_first_in_window(
+                    timestamp_before=feed.queryid_timestamps_list[i - 2],
+                    timestamp_to_check=feed.queryid_timestamps_list[i - 1],
+                    feed_start_timestamp=feed.params.startTime,
+                    feed_window=feed.params.window,
+                    feed_interval=feed.params.interval,
+                )
+                if not in_eligibile_window:
+                    # check price change
+                    feed.queryid_timestamps_list.remove(feed.queryid_timestamps_list[i - 1])
+        return feeds
+
+    def get_real_balance(
+        self, feeds: List[QueryIdandFeedDetails], unclaimed_timestamps_count: Dict[Tuple[bytes, bytes], int]
+    ) -> List[QueryIdandFeedDetails]:
+        # reduce balance based on unclaimed count of reported timestamps
+        for feed in list(feeds):
+            key = (feed.feed_id, feed.query_id)
+            unclaimed_count = unclaimed_timestamps_count[key]
+            feed.params.balance -= feed.params.reward * unclaimed_count
+            # if remaining balance is zero filter out the feed from the dictionary
+            if feed.params.balance <= 0:
+                feeds.remove(feed)
+        return feeds
+
+    async def filter_uneligible_feeds(
+        self, feeds: List[QueryIdandFeedDetails], now_timestamp: TimeStamp
+    ) -> List[QueryIdandFeedDetails]:
+        for feed in list(feeds):
+            # check if your timestamp will be first in window for
+            # this feed if not discard feed_details
+            # lesser node calls to make
+            in_eligible_window, time_diff = self.is_timestamp_first_in_window(
+                feed_interval=feed.params.interval,
+                feed_start_timestamp=feed.params.startTime,
+                feed_window=feed.params.window,
+                timestamp_before=feed.current_value_timestamp,
+                timestamp_to_check=now_timestamp,
+            )
+
+            balance = feed.params.balance
+            price_threshold = feed.params.priceThreshold
+
+            if in_eligible_window:
+                sloped_reward = feed.params.rewardIncreasePerSecond * time_diff
+                feed.params.reward += sloped_reward
+                if balance < feed.params.reward:
+                    feeds.remove(feed)
+                continue
+
+            if price_threshold != 0 and not in_eligible_window:
+                try:
+                    value_before = int(int(feed.current_queryid_value.hex(), 16) / 10**18)
+                except ValueError:
+                    error_status("Error decoding current query id value")
```

Comment:
What if the query's response type isn't a number? That excludes a lot of query types, no?

Reply:
will there be cases where a price threshold could be other than a number?